pnpm/test/switchingVersions.test.ts

```diff
@@ -58,27 +56,3 @@ test('do not switch to pnpm version when a range is specified', async () => {
 
   expect(stdout.toString()).toContain('Cannot switch to pnpm@^9.3.0')
 })
-
-test('throws error if pnpm tools dir is corrupt', () => {
-  prepare()
-  const pnpmHome = path.resolve('pnpm')
-  const env = { PNPM_HOME: pnpmHome }
-  const version = '9.3.0'
-  fs.writeFileSync('.npmrc', 'manage-package-manager-versions=true')
-  writeJsonFile('package.json', {
-    packageManager: `pnpm@${version}`,
-  })
-
-  // Run pnpm once to ensure the tools dir is created.
-  execPnpmSync(['help'], { env })
-
-  // Intentionally corrupt the tool dir.
-  const toolDir = getToolDirPath({ pnpmHomeDir: pnpmHome, tool: { name: 'pnpm', version } })
-  fs.rmSync(path.join(toolDir, 'bin/pnpm'))
-  if (isWindows()) {
-    fs.rmSync(path.join(toolDir, 'bin/pnpm.cmd'))
-  }
-
-  const { stderr } = execPnpmSync(['help'], { env })
-  expect(stderr.toString()).toContain('Failed to switch pnpm to v9.3.0. Looks like pnpm CLI is missing')
-})
```

Comment:
This test is the error in getToolDirPath, and not the error in pnpm CLI missing. So I feel it's better to remove this. Anyway the error is detectable in main.ts.